File: 2325-number-of-ways-to-select-buildings/2325-number-of-ways-to-select-buildings.java
class Solution {
    public long numberOfWays(String s) {
        long ways = 0;
        int zero = 0;
        int ones = 0;
        long zeroOne = 0;
        long oneZero = 0;
        for(int i =0; i < s.length(); i++)
        {
            if(s.charAt(i)=='0'){
                ++zero;
                oneZero += ones;
                ways += zeroOne;
            }
            else
            {
                ++ones;
                zeroOne += zero;
                ways += oneZero;
            }
        }

        return ways;
        
    }
}

File: 2325-number-of-ways-to-select-buildings/2325-number-of-ways-to-select-buildings.py
class Solution:
    def numberOfWays(self, s: str) -> int:
        ways = 0
        one = 0
        zero = 0
        one_zero = 0
        zero_one = 0

        for c in s:
            if c == '0':
                zero += 1
                one_zero += one
                ways += zero_one
            else:
                one += 1
                zero_one += zero
                ways += one_zero
                # print("one", '--', c, '--', zero, '--', one, '--', one_zero, '--', zero_one, '--', ways)
        return ways